src/main/java/datastructs/NumericSample.java

```diff
@@ -2,210 +2,247 @@
-		return this.data_.get(i);
-	}
+    /**
+     * Add the value to the sample
+     */
+    public final void add(Double value) {
 
+        this.data_.add(value);
+        this.falsifyCalculations();
+    }
 
-	/**
-	 * Prints information about the sample
-	 */
-	public void printInfo(){
 
-		getStatistics();
-		this.stats_.printInfo();
-	}
+    /**
+     * Set the i-th entry to the given value
+     */
+    public final void set(int i, Double value) {
 
+        this.data_.set(i, value);
+        this.falsifyCalculations();
+    }
 
-	/**
-	 * Copy the data from the given list
-	 */
-	public void copy(final List<Double> data){
 
-		if(data.size() == 0){
-			throw new IllegalStateException("The input data set has zero size");
-		}
+    /**
+     * Returns the i-th entry of the sample
+     */
+    public final Double get(int i) {
+        return this.data_.get(i);
+    }
 
-		if(this.data_.size() != data.size()){
 
-			// remove data completely
-			this.initialize(data.size());
-		}
+    /**
+     * Prints information about the sample
+     */
+    public void printInfo() {
 
-		Collections.copy(this.data_, data);
-		this.falsifyCalculations();
-	}
+        getStatistics();
+        this.stats_.printInfo();
+    }
 
 
-	/**
-	 * Initialize the sample with zero entries
-	 */
-	protected final void initialize(int size){
+    /**
+     * Copy the data from the given list
+     */
+    public void copy(final List<Double> data) {
+
+        if (data.size() == 0) {
+            throw new IllegalStateException("The input data set has zero size");
+        }
+
+        if (this.data_.size() != data.size()) {
 
-		this.data_ = new ArrayList<Double>(size);
+            // remove data completely
+            this.initialize(data.size());
+        }
 
-		for(int i=0; i< size; ++i){
-			this.data_.add(0.0);
-		}
-	}
+        Collections.copy(this.data_, data);
+        this.falsifyCalculations();
+    }
 
-	/**
-	 * Compute the sample statistics
-	 */
-	protected void compute_sample_statistics(){
 
-		double[] arrayData = ArrayOperations.toArray(this.data_);
-		DescriptiveStatistics stats = new DescriptiveStatistics(arrayData);
+    /**
+     * Initialize the sample with zero entries
+     */
+    protected final void initialize(int size) {
 
-		this.stats_.mean = stats.getMean();
-		this.stats_.variance = stats.getVariance();
-		this.stats_.min = stats.getMin();
-		this.stats_.max = stats.getMax();
-		this.stats_.kurtosis = stats.getKurtosis();
-		this.stats_.skewness = stats.getSkewness();
-		this.stats_.median = new Median().evaluate(arrayData);
-		this.stats_.isValid = true;
-	}
+        this.data_ = new ArrayList<Double>(size);
 
-	protected final void falsifyCalculations(){
+        for (int i = 0; i < size; ++i) {
+            this.data_.add(0.0);
+        }
+    }
+
+    /**
+     * Compute the sample statistics
+     */
+    protected void compute_sample_statistics() {
+
+        double[] arrayData = ArrayOperations.toArray(this.data_);
```

Comment:
We expose an ```asArray``` function so we don't need the ```ArrayOperations``` here.

Reply:
I think we have to merge create_visualization_classes into this branch.
maybe these are left from an earlier implementation.